File: .github/tests/python/test_installation.py
#!/usr/bin/env python3
"""
Installation test for SLIM Python bindings.

This test verifies that the slim_bindings package can be imported
and initialized successfully.
"""

import slim_bindings


def main():
    print("SLIM Python Bindings Installation Test")
    print("=" * 50)

    # Initialize SLIM (required before any operations)
    slim_bindings.initialize_with_defaults()
    print("SLIM initialized successfully")

    print("Installation test passed!")


if __name__ == "__main__":
    main()

File: .github/tests/go/test_installation.go
package main

import (
	"fmt"
	"os"

	slim "github.com/agntcy/slim-bindings-go"
)

// Installation test for SLIM Go bindings.
//
// This test verifies that the slim_bindings package can be imported
// and initialized successfully.
func main() {
	fmt.Println("🚀 SLIM Go Bindings Installation Test")
	fmt.Println("==================================================")

	// Initialize SLIM (required before any operations)
	slim.InitializeWithDefaults()
	fmt.Println("✅ SLIM initialized successfully")

	fmt.Println("✅ Installation test passed!")
	os.Exit(0)
}